lib/sles4sap/sap_deployment_automation_framework/deployment_connector.pm

```diff
@@ -360,10 +361,11 @@ permanent SDAF infrastructure.
 sub destroy_orphaned_resources {
     my (%args) = @_;
     $args{timeout} //= 1200;
-    # List all resources containing 'deployment_id' tag - this should filter out only resources created by an openQA test.
+    # List all resources containing 'deployment_id' tag with exception of ones containing 'no cleanup' tag
+    # Result will show only resources created by OpenQA tests and only those which are allowed to be cleaned up.
     my $all_resources = az_resource_list(
         resource_group => get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP'),
-        query => '[?tags.deployment_id].{resource_id:id, creation_time:createdTime}'
+        query => '[?tags.deployment_id && tags.' . no_cleanup_tag() . '].{resource_id:id, creation_time:createdTime}'
```

Comment:
Which test steps in the VRs can verify this query works?

Reply:
That is a good question :D Problem is that any current test without the changes will still delete this deployment. I guess the easiest will be to merge and wait till tomorrow. I plan to run another Vr in the evening, but if there is another job running in between it will just destroy the deployment